pkg/hookbot/hookbot.go

```diff
@@ -82,6 +84,18 @@ func New(key string) *Hookbot {
 	return h
 }
 
+// BothPubSub is an endpoint which supports either publishing or subscribing.
+// If it is a POST request, it is publishing, otherwise it is subscribing.
+func (h *Hookbot) BothPubSub(pub, sub http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "POST" {
+			pub.ServeHTTP(w, r)
+			return
+		}
+		sub.ServeHTTP(w, r)
```

Comment:
I did originally consider it but thought I would be happy with whatever logic was in the sub. Updated anyway.